src/org/apache/commons/httpclient/HttpMethodBase.java

```diff
@@ -1368,6 +1372,84 @@ protected void addCookieRequestHeader(HttpState state, HttpConnection conn)
     }
 
     /**
+     * Put all the cookies in a single header line. 
+     * 
+     * Merge the cookies already present in the request 
+     * with the cookies coming from the state. 
+     * 
+     * @param state the {@link HttpState state} information associated with this method
+     * @param conn the {@link HttpConnection connection} used to execute
+     *        this HTTP method
+     * @param cookies associated with the {@link HttpState state}
+     * 
+     */
+	private void putAllCookiesInASingleHeader(HttpState state, HttpConnection conn,
+			Cookie[] cookies) {
+		
+		LOG.trace("enter putAllCookiesInASingleHeader(HttpState state, HttpConnection conn, Cookie[] cookies)" );
+		
+		String host = getHost(state, conn);
+		CookieSpec matcher = getCookieSpec(state);
+		//use a map to make sure we only have one cookie per name
+		HashMap<String, Cookie> mergedCookies = new HashMap<String, Cookie>();		
+		Header[] cookieheaders = getRequestHeaderGroup().getHeaders("Cookie");
+		for (int i = 0; i < cookieheaders.length; i++) {
```

Comment:
It might be cleaner to use an enhanced for here `for (Header cookieheader : cookieheaders) {`